pkg/handlers/handlers.go

```diff
@@ -78,51 +58,43 @@ const (
 // * HTTP 409 if a previous profiling is still ongoing,
 // * HTTP 200 if the agent is ready
 func (h *Handlers) Status(w http.ResponseWriter, r *http.Request) {
-	if h.errorFileExists() {
-		uid, err := readUIDFromFile(filepath.Join(h.StorageFolder, "agent."+string(errorFile)))
+	id, state, err := h.turnTaker.WhoseTurn()
+	if err != nil {
+		http.Error(w, "error retrieving service status",
+			http.StatusInternalServerError)
+		hlog.Errorf("Error retrieving service status : %v", err)
+		return
+	}
+	switch state {
+	case turntaker.InError:
+		err = respondBusyOrError(id.String(), w, r, true)
 		if err != nil {
-			http.Error(w, "unable to read error file",
+			http.Error(w, "unable to send response",
 				http.StatusInternalServerError)
-			hlog.Error("Unable to read error file")
+			hlog.Error("unable to send response")
```

Comment:
Define a constant instead of duplicating this literal 8 times (unable to send response)

Reply:
Done